Jak-Sidious/AndroidLevelUp
# Ui/ux revamp share functionality readme 156037248

#### What does this PR do?

1. Create the share functionality
2. Implement Swipe to Refresh Layout
3. Implement Progress bar
4. Update ReadME
5. Create Network Utility class
6. Create SnackBar
7. Implement Unit Tests
8. Implement UI Test

#### Description of Task to be completed?

- Enabled the user with access to the application to share a github users details
- Implement the swipe to refresh where when a user swipes down fron tge top, the api call is refreshed accordingly
- A progress bar was created to cater to the pulling of data from the api.
- Readme was updated to include details on installation of the app, how to test as well as information about the Author.
- Creation of a network utility class to cater to the application having or not having an internet connection and the accompanying snackbar to show the error message.
- Creation of Both Unit and UI tests for the application as a proof of concept.

#### How should this be manually tested?

1. To manually test this, the user can clone the repo and do the following to test the functionality
2. Start the application and click on any user after which they can click on the black share button in the bottom right corner to open the share screen.
3. On the main page, the user can swipe down to witness the swipe to refresh as well as view the progress bar.
4. In the repository, the user can view the Readme with the detailed information.
5. The user can disable the mobile data on the application and run it to check if the network utility is properly running.
6. To run the tests the user can run the command 
`./gradlew createDebugCoverageReport`
After which the user can open the acess the following directories to get the coverage report
`app/build/coverage` under the files index.html and jacoco-sessions.html
#### Any background context you want to provide?
none
#### What are the relevant pivotal tracker stories?
#156008359
#156037248
#156185411
#156187450
#156185242
#157354404

#### Screenshots (if appropriate)
Sharing
![sharing](https://user-images.githubusercontent.com/30991429/41113897-8e771708-6a8b-11e8-910b-ab78cde95171.gif)

Refreshing
![refresh](https://user-images.githubusercontent.com/30991429/41114046-ea06d630-6a8b-11e8-98eb-8a40e4a22756.gif)

Network Utility
![network](https://user-images.githubusercontent.com/30991429/41114124-2c9d0938-6a8c-11e8-9b43-b4d16e764d40.gif)


#### Questions:
